api/fbs/v1/query.fbs

```diff
@@ -0,0 +1,102 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace v1;
+
+enum BinaryOp: byte {
+    EQ,
+    NE,
+    LT,
+    GT,
+    LE,
+    GE,
+    HAVING,
+    NOT_HAVING,
+    RANGE
+}
+
+table ByteValue {
+    value: [ubyte];
+}
+
+table Tag {
+    key: [ubyte];
+    values: [ByteValue];
+}
+
+table TagQuery {
+    tag: Tag;
+    op: BinaryOp;
+}
+
+enum QueryOrder : byte {
+    ByStartTime,
+    ByDuration
+}
+
+enum Sort : byte {
+    DESC,
+    ASC
+}
+
+table Entity {
+    entity_id: string;
+    trace_id: string;
+    start_time: uint64;
+    end_time: uint64;
+    duration: uint64;
+    data_binary: [ubyte];
+    fields: [Tag];
+    tags: [Tag];
+}
+
+table BasicTrace {
+    entity_id: string;
+    endpoint_names: [string];
+    duration: uint64;
+    start_time: uint64;
+    trace_ids: [string];
+}
+
+table TraceBrief {
+    traces: [BasicTrace];
+    total: int32;
+}
+
+table TraceQueryCriteria {
+    trace_id: string;
+    start_time_nanoseconds: uint64;
+    end_time_nanoseconds: uint64;
+    limit: uint32;
+    query_order: QueryOrder = ByStartTime;
+    sort: Sort = DESC;
+    tags: [TagQuery];
```

Comment:
Tag is not searchable.

Reply:
As I've discussed with @hanahmily, even if the tags are not indexed, it is still possible to search by post-processing.

Reply:
No, we shouldn't. Trace is a huge data set, there is not a chance to query a billion-level dataset with all rows scaning.

Reply:
This is an operation/maintenance purposed field. An operator or user can filter traces based on any tags and fields. 

The potential high cardinality issue as you mentioned still exists. The user could use other indexed fields to reduce the dataset. In the future, banyandb will provides some analysis data to show the candidate number, which will help user to improve their query criteria.

Reply:
Let's review the real requirements, there is never a case of tags. If you think there is, please put an example here. No matter SkyWalking or Zipkin query, this is not necessary.